lib/newProfile.ml

```diff
@@ -92,9 +92,77 @@ let gettimeopt = function
 let prtime t =
   Format.sprintf "%.0f" (t *. 1E6)
 
-let global_start = gettime()
-let global_start_stat = Gc.quick_stat()
-let global_start_count = Ok Int64.zero
+module Counters = struct
+
+  (* time is handled separately because it has special status in the output format *)
+  type t = {
+    major_words : float;
+    minor_words : float;
+    major_collections : int;
+    minor_collections : int;
+    minor_time : int64 (* nanoseconds *);
+    major_time : int64 (* nanoseconds *);
+    instr : System.instruction_count;
+  }
+
+  let current_minor_time = ref Int64.zero
+  let current_major_time = ref Int64.zero
```

Comment:
Maybe instead of treating them as counters we should be emitting enter/leave events? I'm a bit worried that we may gc while printing another event which would cause bad output.

Reply:
>I'm a bit worried that we may gc while printing another event which would cause bad output.

actually it's polling based not async so that shouldn't be a problem
OTOH translating the runtime_events timestamps to times that match the other events well enough and keeping things well backeted is more troublesome

Reply:
I think my mental model of GC and runtime_events was off, cf new commit.